test/extensions/transport_sockets/tls/handshaker_test.cc

```diff
@@ -163,6 +165,100 @@ TEST_F(HandshakerTest, ErrorCbOnAbnormalOperation) {
   EXPECT_EQ(post_io_action, Network::PostIoAction::Close);
 }
 
+class HandshakerWithOutOfProcessComponent : public Ssl::Handshaker {
+ public:
+  void initialize(SSL&) override{};
+
+  Network::PostIoAction doHandshake(
+      Ssl::SocketState &state, SSL*,
+      Ssl::HandshakerCallbacks &callbacks) override {
+    Network::PostIoAction action;
+    auto thread = Thread::threadFactoryForTest().createThread([&] {
+      callbacks.HandBack(out_of_process_component.doHandshake(
+          callbacks.HandOff(), state, callbacks, *transport_socket_callbacks_,
+          action));
+    });
+    thread->join();
```

Comment:
Can you make this more representative by not waiting for the resposne in `doHandshake()`, but instead having the thread post a message back that makes envoy resume processing? This test is simulating a blocking-call to the out-of-thread component, which isn't correct.

Reply:
I've since removed this test case.